app/api/events.py

```diff
@@ -586,7 +619,229 @@ async def confirmation(request: Request, id: str, token: AccessTokenPayload = De
+        await penalize(db, m_id)
+        # Add id to registeredPenalties on event
+        updates.append(UpdateOne(
+            {"eid": event["eid"]}, 
+            {"$addToSet": {"registeredPenalties": m_id}}
+        ))
 
+    # Write to event db
+    res = db.events.bulk_write(updates)
+
+    if not res:
+        raise HTTPException(500)
+
+
+
+@router.post('/{id}/qr', dependencies=[Depends(validate_uuid)])
+def create_registration_qr(request: Request, id: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    """ Generate a registration QR code for event """
+    db = get_database(request)
+    event = get_event_or_404(db, id)
+
+    # Check whether QR already has been created
+    rid_created = 'register_id' in event and event['register_id']
+    filepath = f'{get_qr_path(request)}/{event["eid"].hex}.pdf'
+    if rid_created and os.path.exists(filepath):
+        raise HTTPException(400, "Event QR already created")
+    
+
+    # Generate new unique id for attendence registration
+    # Only if rid has not yet been created
+    register_id = uuid4()
+
+    res = db.events.update_one(
+        {'eid': UUID(id)},
+        {'$set': {'register_id': register_id}}
+    )
+
+    if not res:
+        raise HTTPException(500)
+    
+    # Create directory if not present
+    if not os.path.exists(get_qr_path(request)):
+        os.makedirs(get_qr_path(request))
+
+    # Generate qr code 
+    url = f'https://td-uit.no/event/{register_id}/register/'
+    img = qr.make(url)
+    qr_path = get_qr_path(request)
+    img_path = f'{qr_path}/{register_id.hex}.png'
+    img.save(img_path)
+
+    # Create PDF
+    pdf = FPDF()
+    pdf.add_page()
+
+    # Add title
+    title = event['title']
+    title_h = pdf.h * 0.05
+    pdf.set_font('helvetica', size=26)
+    pdf.multi_cell(0, title_h, txt = title, align = 'C', max_line_height=title_h)
+
+    # Get width and heigh to display image
+    w, h = img.size
+    pdf_w = pdf.w * 0.8
+    pdf_h = pdf_w * h / w
+    y = title_h + pdf.h * 0.08
+
+    # Add QR code
+    pdf.image(img_path, x = pdf.w * 0.1, y = y , w = pdf_w, h = pdf_h)
+
+    # Add new line of text below the image
+    pdf.set_xy(0, y + pdf_h + pdf.h * 0.05)
+    pdf.set_font('helvetica', size=10)
+    pdf.cell(0, 20, txt=url, align='C')
+
+    # Save PDF
+    pdf.output(f'{qr_path}/{event["eid"].hex}.pdf')
+
+    # Clean up QR image
+    os.remove(img_path)
+
+
+@router.get('/{id}/qr', dependencies=[Depends(validate_uuid)])
+def get_registration_qr(request: Request, id: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    """ Get qr code link to registration page of event """
+    db = get_database(request)
+    event = get_event_or_404(db, id)
+
+    # Check registration id
+    if 'register_id' not in event:
+        raise HTTPException(400, "Event not open for registration")
+    
+    if not event['register_id']:
+        raise HTTPException(400, "Event not open for registration")
```

Comment:
can this be merged into one?

Reply:
Not sure how this works in python. If these conditions are in the same if statement, will it not produce a keyerror? If not, they can be moved to one statement.

Reply:
get_event_or_404 should parse the event `EventDB.parse_obj(event)` so event["register_id"] should be None

Reply:
In that case, do I need the first if statement at all?

Reply:
I don't tink so, but it should be tested as i am not 100% sure :laughing: